src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueTests.cs

```diff
@@ -7183,5 +7183,934 @@ static void F()
+  IL_0016:  ldloc.0
+  IL_0017:  ldfld      ""(int, int) System.ValueTuple<int, (int, int)>.Item2""
+  IL_001c:  ldfld      ""int System.ValueTuple<int, int>.Item1""
+  IL_0021:  add
+  IL_0022:  ldloc.0
+  IL_0023:  ldfld      ""(int, int) System.ValueTuple<int, (int, int)>.Item2""
+  IL_0028:  ldfld      ""int System.ValueTuple<int, int>.Item2""
+  IL_002d:  add
+  IL_002e:  stloc.1
+  IL_002f:  br.s       IL_0031
+  IL_0031:  ldloc.1
+  IL_0032:  ret
+}
+");
+
+            var md0 = ModuleMetadata.CreateFromImage(v0.EmittedAssemblyData);
+
+            var generation0 = EmitBaseline.CreateInitialBaseline(md0, v0.CreateSymReader().GetEncMethodDebugInfo);
+            var diff1 = compilation1.EmitDifference(
+                generation0,
+                ImmutableArray.Create(
+                    new SemanticEdit(SemanticEditKind.Update, f0, f1, GetSyntaxMapFromMarkers(source0, source1), preserveLocalVariables: true)));
+
+            diff1.VerifyIL("C.F", @"
+{
+  // Code size       36 (0x24)
+  .maxstack  4
+  .locals init ([unchanged] V_0,
+                [int] V_1,
+                System.ValueTuple<int, int, int> V_2, //x
+                int V_3)
+  IL_0000:  nop
+  IL_0001:  ldloca.s   V_2
+  IL_0003:  ldc.i4.1
+  IL_0004:  ldc.i4.2
+  IL_0005:  ldc.i4.3
+  IL_0006:  call       ""System.ValueTuple<int, int, int>..ctor(int, int, int)""
+  IL_000b:  ldloc.2
+  IL_000c:  ldfld      ""int System.ValueTuple<int, int, int>.Item1""
+  IL_0011:  ldloc.2
+  IL_0012:  ldfld      ""int System.ValueTuple<int, int, int>.Item2""
+  IL_0017:  add
+  IL_0018:  ldloc.2
+  IL_0019:  ldfld      ""int System.ValueTuple<int, int, int>.Item3""
+  IL_001e:  add
+  IL_001f:  stloc.3
+  IL_0020:  br.s       IL_0022
+  IL_0022:  ldloc.3
+  IL_0023:  ret
+}
+");
+
+            var diff2 = compilation2.EmitDifference(
+                diff1.NextGeneration,
+                ImmutableArray.Create(
+                    new SemanticEdit(SemanticEditKind.Update, f1, f2, GetSyntaxMapFromMarkers(source1, source2), preserveLocalVariables: true)));
+
+            diff2.VerifyIL("C.F", @"
+{
+  // Code size       32 (0x20)
+  .maxstack  3
+  .locals init ([unchanged] V_0,
+                [int] V_1,
+                [unchanged] V_2,
+                [int] V_3,
+                System.ValueTuple<int, int> V_4, //x
+                int V_5)
+  IL_0000:  nop
+  IL_0001:  ldloca.s   V_4
+  IL_0003:  ldc.i4.1
+  IL_0004:  ldc.i4.3
+  IL_0005:  call       ""System.ValueTuple<int, int>..ctor(int, int)""
+  IL_000a:  ldloc.s    V_4
+  IL_000c:  ldfld      ""int System.ValueTuple<int, int>.Item1""
+  IL_0011:  ldloc.s    V_4
+  IL_0013:  ldfld      ""int System.ValueTuple<int, int>.Item2""
+  IL_0018:  add
+  IL_0019:  stloc.s    V_5
+  IL_001b:  br.s       IL_001d
+  IL_001d:  ldloc.s    V_5
+  IL_001f:  ret
+}
+");
+        }
+
+        [Fact]
+        public void Tuple_DecompositionIgnoreWithVariableDeclaration()
```

Comment:
>Tuple_DecompositionIgnoreWithVariableDeclaration [](start = 20, length = 48)

I would also delete this test - it's just adding names to tuple elements. The local types/IL structure is the same as the test above.